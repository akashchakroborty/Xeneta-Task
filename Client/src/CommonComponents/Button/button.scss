@import '../../App/app.variables.scss';

.button {
  background: $primary-color;
  border-radius: 2em;
  border-color: $primary-color;
  &:hover,
  &:focus {
    background: $white-color;
    color: $primary-color;
    border-radius: 2em;
    border-color: $primary-color;
  }
}
