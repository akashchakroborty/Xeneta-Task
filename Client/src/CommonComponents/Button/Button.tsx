import React from 'react';
import { Button } from 'antd';

import './button.scss';

export interface ButtonProps {
  type: 'text' | 'link' | 'ghost' | 'default' | 'primary' | 'dashed' | undefined;
  onClick: () => void;
  value: string;
  disabled?: boolean;
}

const ButtonComponent: React.FC<ButtonProps> = ({ type, onClick, value, disabled = false }) => {
  return (
    <Button type={type} onClick={onClick} className="button" disabled={disabled}>
      {value}
    </Button>
  );
};

export default ButtonComponent;
