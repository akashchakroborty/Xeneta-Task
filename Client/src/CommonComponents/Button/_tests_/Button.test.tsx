import React from 'react';
import { shallow } from 'enzyme';
import Button, { ButtonProps } from '../Button';

describe('should test the Button Component', () => {
  const mockProps: ButtonProps = {
    type: 'text',
    onClick: jest.fn(),
    value: 'Mock Button',
    disabled: false,
  };
  test('should render the Component', () => {
    const component = shallow(<Button {...mockProps} />);
    expect(component).toMatchSnapshot();
  });
});
