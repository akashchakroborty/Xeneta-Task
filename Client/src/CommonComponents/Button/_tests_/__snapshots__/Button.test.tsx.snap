// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the Button Component should render the Component 1`] = `
<Button
  className="button"
  disabled={false}
  onClick={[MockFunction]}
  type="text"
>
  Mock Button
</Button>
`;
